native/src/command/action.rs

```diff
@@ -0,0 +1,24 @@
+use crate::clipboard;
+use crate::window;
+
+pub enum Action<T> {
+    Future(iced_futures::BoxFuture<T>),
```

Comment:
for the future, I suggest adding `Action::Stream(futures:: stream:: Box Stream<T>)` for animation and for download_progress

Reply:
but what if I want to use tokio's async-stream? async-stream generates a state structure. how to hash them?

it seems to me that it would be easier to create streams this way
```
Command::stream(stream! {
... 
yield Message::Progress(..) ;
})
```

Reply:
Commands cannot be canceled while subscriptions can.

I know it'd be more convenient, but convenience has a cost in this case.